owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -159,9 +159,17 @@ class MainApp : Application() {
         initDependencyInjection()
     }
 
-    fun startLogsIfEnabled() {
-        enabledLogging = PreferenceManager.getDefaultSharedPreferences(applicationContext)
-            .getBoolean(SettingsLogsFragment.PREFERENCE_ENABLE_LOGGING, false)
```

Comment:
In my point of view this is complete overengineered, a simple 

        enabledLogging = BuildConfig.DEBUG || PreferenceManager.getDefaultSharedPreferences(applicationContext)
             .getBoolean(SettingsLogsFragment.PREFERENCE_ENABLE_LOGGING, false) 

is enough to have logging during debug.

Reply:
If we do it like this, the switch in the UI for the "Enable logging" preference won't have a consistent behaviour with the actual state of logging.